frontend/packages/dev-console/src/components/catalog/service/CatalogServiceProvider.tsx

```diff
@@ -34,23 +34,31 @@ const CatalogServiceProvider: React.FC<CatalogServiceProviderProps> = ({
     extensionsResolved,
   ] = useCatalogExtensions(catalogType);
 
-  const [catalogItemsMap, setCatalogItemsMap] = React.useState<{ [key: string]: CatalogItem[] }>(
-    {},
-  );
+  const [extItemsMap, setExtItemsMap] = React.useState<{ [uid: string]: CatalogItem[] }>({});
   const [loadError, setLoadError] = React.useState();
 
   const loaded =
     extensionsResolved &&
-    (catalogTypeExtensions.length === 0 ||
-      catalogTypeExtensions.every(({ properties: { type } }) => catalogItemsMap[type]));
+    (catalogProviderExtensions.length === 0 ||
+      catalogProviderExtensions.every(({ uid }) => extItemsMap[uid]));
 
-  const catalogItems = React.useMemo(
-    () => (loaded ? _.flatten(Object.values(catalogItemsMap)) : []),
-    [catalogItemsMap, loaded],
-  );
+  const catalogItems = React.useMemo(() => {
+    if (!loaded) {
+      return [];
+    }
+    const itemMap = _.flatten(catalogProviderExtensions.map((e) => extItemsMap[e.uid])).reduce(
+      (acc, item) => {
+        acc[item.uid] = item;
+        return acc;
+      },
+      {} as { string: CatalogItem },
```

Comment:
```suggestion
      {} as { [key: string]: CatalogItem },
```

What you have says you'll have a key `string` equal to `CatalogItem`... which I'm sure is not what you're looking for :)

Reply:
Updated.
Surprisingly it was not a TS error...